docs/language/informal/toplevel-inference.md

```diff
@@ -0,0 +1,338 @@
+﻿# Toplevel inference
+
+Owner: leafp@google.com
+
+Status: Proposal
+
+
+## Top level inference overview
+
+Top level inference has two phases:
+
+1. **Method override inference**
+    * If you omit a return type or parameter type from an overridden method,
```

Comment:
There are fairly complex rules about methods overriding in the specification. I try to understand whether you indent that we follow all these rules, or whether we use simple 'find by name and infer formal parameter types one-by-one'.